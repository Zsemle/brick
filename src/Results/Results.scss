@import '../style/layout.scss';

.results {
  list-style: none;
  padding: 0;

  @media (min-width: $bp-medium) {
    margin: 0;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    column-gap: $space-small;
    row-gap: $space-small;
  }
}
