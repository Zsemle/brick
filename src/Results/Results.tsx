import React, { FunctionComponent } from 'react'
import ResultItem from '../ResultItem/ResultItem'
import { Experience } from '../types/brick-types'
import './Results.scss'

export interface ResultsProps {
  experiences: Experience[] | null
}

const T:any = {
  noResults: 'No results match your criteria. Try changing the filter options.'
}

const Results:FunctionComponent<ResultsProps> = ({ experiences }:ResultsProps):JSX.Element => {
  if (experiences && experiences.length > 0) {
    return (
      <ul className="results">
        {experiences?.map((experience, key) => <ResultItem
          name={experience.name}
          description={experience.description}
          category={experience.category}
          price={experience.price}
          key={experience.name + key}
        />)}
      </ul>
    )
  }
  return <p>{T.noResults}</p>
}

export default Results
