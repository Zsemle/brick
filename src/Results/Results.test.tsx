import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import Results, { ResultsProps } from './Results'

describe('the results component', () => {
  const props:ResultsProps = {
    experiences: [
      {
        name: 'test experience 1',
        description: 'description of the experience',
        category: 'brain games',
        price: {
          amount: 234,
          currency: 'USD'
        }
      },
      {
        name: 'test experience 2',
        description: 'description of the second experience',
        category: 'brain games',
        price: {
          amount: 222222,
          currency: 'GBP'
        }
      }
    ]
  }

  const wrapper:ShallowWrapper = shallow(<Results {...props}/>)

  it('renders the component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the component when no experiences are available', () => {
    wrapper.setProps({
      experiences: []
    })
    expect(wrapper).toMatchSnapshot()
  })
})
