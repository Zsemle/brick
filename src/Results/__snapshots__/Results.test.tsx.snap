// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the results component renders the component 1`] = `
<ul
  className="results"
>
  <ResultItem
    category="brain games"
    description="description of the experience"
    key="test experience 10"
    name="test experience 1"
    price={
      Object {
        "amount": 234,
        "currency": "USD",
      }
    }
  />
  <ResultItem
    category="brain games"
    description="description of the second experience"
    key="test experience 21"
    name="test experience 2"
    price={
      Object {
        "amount": 222222,
        "currency": "GBP",
      }
    }
  />
</ul>
`;

exports[`the results component renders the component when no experiences are available 1`] = `
<p>
  No results match your criteria. Try changing the filter options.
</p>
`;
